hedging/tests/fixtures/scenario_01.json

```diff
@@ -0,0 +1,40 @@
+{
+    "payloads": [
```

Comment:
I don't really understand this test. What edge cases / code paths is it going through that the `hedging` test didn't touch?
All things being equal we should prefer the test with the shortest runtime. I'm wondering if this is actually redundant and we keep the original one - or perhaps extend it slightly if there is a code path we missed.

Note that integration tests are not intended to test the exact logic - that's what unit tests are for.

Reply:
All right, in that case I will restore your test and remove this one and we should be good to go.